cmd/server/helper_keys.go

```diff
@@ -18,7 +18,7 @@ func createRS256KeysIfNotExist(c *config.Config, set, kid, use string) {
 	if _, err := ctx.KeyManager.GetKey(set, kid); errors.Cause(err) == pkg.ErrNotFound {
 		logrus.Infof("Key pair for signing %s is missing. Creating new one.", set)
 
-		keys, err := generator.Generate("")
+		keys, err := generator.Generate(set)
```

Comment:
So that could work, but when I add the `kid` to the JWT header, which would need to happen here https://github.com/ory/hydra/blob/master/oauth2/consent_strategy.go#L102 I would need to know what that ID is. We could have the understanding that the `kid` in the JWT header is appended with a known value -- say a hash of the set name, and then when we send the keys in the `well-known` response we append that as well. 

The downside of this is that if a person looks at the key in `well-known` and tries to make an API call to hydra modifying it, that `kid` it wouldn't exist. The same could be said for writing policy.

A bit of a rock and a hard place 😄